.github/actions/install-go-and-dependencies/action.yml

```diff
@@ -0,0 +1,18 @@
+name: InstallGo 
+description: 'Installs Go Downloads'
+runs:
+  using: "composite"
```

Comment:
Using a composite action can make workflow code more modular and easier to maintain. Instead of repeating the same set of steps across multiple workflows, we can define a composite action once and then we reference it wherever it's needed,Here the i used this action for setup go and used as a single line ,I will use the same action for the next workflows that are used to do test coverage for kit guest cluster and Eks guest cluster.